cosmos/providers/databricks/notebook.py

```diff
@@ -0,0 +1,199 @@
+from __future__ import annotations
+
+import time
+from typing import Any
+
+from airflow.exceptions import AirflowException
+from airflow.models.operator import BaseOperator
+from airflow.providers.databricks.hooks.databricks import DatabricksHook
+from airflow.utils.context import Context
+from databricks_cli.runs.api import RunsApi
+from databricks_cli.sdk.api_client import ApiClient
+
+
+class DatabricksNotebookOperator(BaseOperator):
+    """
+    Launches a notebook to databricks using an Airflow operator.
+
+    The DatabricksNotebookOperator allows users to launch and monitor notebook deployments on Databricks as Aiflow tasks.
+    It can be used as a part of a DatabricksWorkflowTaskGroup to take advantage of job clusters,
+    which allows users to run their tasks on cheaper clusters that can be shared between tasks.
+
+    Here is an example of running a notebook as a part of a workflow task group:
+
+    .. code-block: python
+
+        with dag:
+            task_group = DatabricksWorkflowTaskGroup(
+                group_id="test_workflow",
+                databricks_conn_id="databricks_conn",
+                job_clusters=job_cluster_spec,
+                notebook_params=[],
+            )
+            with task_group:
+                notebook_1 = DatabricksNotebookOperator(
+                    task_id="notebook_1",
+                    databricks_conn_id="databricks_conn",
+                    notebook_path="/Users/daniel@astronomer.io/Test workflow",
+                    source="WORKSPACE",
+                    job_cluster_key="Shared_job_cluster",
+                )
+                notebook_2 = DatabricksNotebookOperator(
+                    task_id="notebook_2",
+                    databricks_conn_id="databricks_conn",
+                    notebook_path="/Users/daniel@astronomer.io/Test workflow",
+                    source="WORKSPACE",
+                    job_cluster_key="Shared_job_cluster",
+                    notebook_params={
+                        "foo": "bar",
+                    },
+                )
+                notebook_1 >> notebook_2
+
+
+        :param notebook_path: the path to the notebook in Databricks
+        :param source: Optional location type of the notebook. When set to WORKSPACE, the notebook will be retrieved
+        from the local Databricks workspace. When set to GIT, the notebook will be retrieved from a Git repository
+        defined in git_source. If the value is empty, the task will use GIT if git_source is defined
+        and WORKSPACE otherwise. For more information please visit
+        https://docs.databricks.com/dev-tools/api/latest/jobs.html#operation/JobsCreate
+        :param databricks_conn_id: the connection id to use to connect to Databricks
+        :param notebook_params: the parameters to pass to the notebook
+        :param kwargs:
+    """
+
+    template_fields = ("databricks_run_id",)
```

Comment:
WDYT of having the `notebook_path` as a template field as well?

Reply:
I discussed this with @dimberman  and he suggested, for now, we don't introduce more files - and that if users need, we can add more fields to template fields.